pkg/distributor/push.go

```diff
@@ -187,3 +189,10 @@ func toHTTPStatus(ctx context.Context, pushErr error, limits *validation.Overrid
 
 	return http.StatusInternalServerError
 }
+
+func addHeaders(w http.ResponseWriter, err error) {
```

Comment:
@ying-jeanne this can be useful in your retry-after work :)

Reply:
In fact, after we had a call today I added the `addHeaders` method to make @ying-jeanne's work easier.